server/svix-server/src/v1/endpoints/message.rs

```diff
@@ -245,12 +254,28 @@ async fn create_message(
         cache,
         ..
     }): State<AppState>,
-    ValidatedQuery(CreateMessageQueryParams { with_content }): ValidatedQuery<
-        CreateMessageQueryParams,
-    >,
-    permissions::OrganizationWithApplication { app }: permissions::OrganizationWithApplication,
+    Path(ApplicationPath { app_id }): Path<ApplicationPath>,
+    ValidatedQuery(CreateMessageQueryParams {
+        with_content,
+        ignore_missing_app,
+    }): ValidatedQuery<CreateMessageQueryParams>,
+    permissions::Organization { org_id }: permissions::Organization,
     ValidatedJson(data): ValidatedJson<MessageIn>,
-) -> Result<(StatusCode, Json<MessageOut>)> {
+) -> Result<(StatusCode, axum::response::Response)> {
+    let app = match ctx!(
```

Comment:
Nevermind, ignore me. I read too quickly.

Reply:
Just to clarify, I missed that we were already querying the app in `permissions::OrganizationWithApplication`. So I think what you did is fine.

It's still _probably_ better to rely on the lookup from `layered_fetch`, you'll just have to change it to accept an `ApplicationIdOrUid` instead of an `ApplicationId`. Maybe not too bad of a change but I also understand if you don't want to do that since it's out of scope.